src/Meilisearch/Errors/MeilisearchApiError.cs

```diff
@@ -0,0 +1,26 @@
+namespace Meilisearch
+{
+    using System;
+
+    /// <summary>
+    /// Error sent by MeiliSearch API.
+    /// </summary>
+    public class MeilisearchApiError : Exception
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MeilisearchApiError"/> class.
+        /// Handler Exception received by Meilisearch API with inner exception.
+        /// </summary>
+        /// <param name="apiError">Specific error message from Meilisearch Api.</param>
+        public MeilisearchApiError(MeilisearchApiErrorContent apiError)
+            : base(string.Format("MeilisearchApiError, Message: {0}, ErrorCode: {1}, ErrorType: {2}, ErrorLink: {3}", apiError.Message, apiError.ErrorCode, apiError.ErrorType, apiError.ErrorLink))
+        {
+            this.Content = apiError;
+        }
```

Comment:
I didn't check but I'm worried the MeiliSearchApiError will not work if the body does not contain the `errorCode`, `errorType` ...
Indeed, it could happen in the case NGINX catches the error before MeiliSearch and returns a 502, for example -> the body will not be formatted as expected and it will raise an exception in your exception.

You should add a test like I did in the dart SDK with a wrong URL -> the MeiliSearch server will not return the "nice" mody with all the parameters: https://github.com/meilisearch/meilisearch-dart/blob/b9f0526736b7fc9baae8781a4fc5741212970698/test/exceptions_test.dart#L21-L29

Reply:
You right I change the code to catch this possibility and add a test.